pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -52,6 +46,17 @@ const (
 	InvitationID = "invitationID"
 )
 
+// didCommChMessage type to correlate actionEvent message(go channel) with callback message(internal go channel).
+type didCommChMessage struct {
+	ID              string
+	DIDCommCallback dispatcher.DIDCommCallback
+}
+
+// ConnectionRecord contain info about did exchange connection
+type ConnectionRecord struct {
```

Comment:
Sudesh has created separate file connections, do you think this struct belong there or here .. ?

Reply:
@sudeshrshetty @fqutishat ^^^

Reply:
Yes I will move it

Reply:
A couple thoughts:
- As store is provided to Client, it would make more sense (to me) to have a set of protocol methods that can marshal/unmarshal using a store and protocol record types. I don't like the idea of unnecessary methods in the service package. These methods could live in a persistence sub-package as Sudesh started.
- "QueryFooByID" seems a bit verbose.
  - Retrieving a record by ID is the default mechanism (ByID is not really needed).
  - If we have a bunch of way to do queries, it might make sense to have a criteria object passed to a "QueryFoo" method. As such, we probably should use a different prefix for the default mechanism (other than Query).

Reply:
I think this applies here: https://golang.org/doc/effective_go.html#Getters

Reply:
One other thought: the "Record" suffix is unnecessary

Reply:
done